server/controllers/AutomationController.js

```diff
@@ -448,43 +474,74 @@ const finalizeTestPlanReportIfAllTestsMatchHistoricalResults = async ({
     );
     if (!historicalResults) return;
 
+    let differentResponsesCount = 0;
+    let allOutputsMatch = true;
+
     // Validate each applicable test's results against historical results
     for (const test of applicableTests) {
       const currResult = testPlanRun.testResults.find(
         tr => String(tr.testId) === String(test.id)
       );
+
+      if (!currResult) continue;
```

Comment:
Don't we want to set `allOutputsMatch = false` in this case? And if so maybe we can consider this branch an early optimization and just rely on the condition that follows the calculation of `histResult`.

Reply:
Good callout! Addressed in [e173a1f](https://github.com/w3c/aria-at-app/pull/1441/commits/e173a1febd328013a7c63765071404bff875ae68)